### PR TITLE
SDK should not cache data with FetchOption::OnlineOnly requests.

### PR DESCRIPTION
**Pull request by LiubovDidkivska**
Add Check if fetch_option is not OnlineOnly, and cache only in that case.

Resolves:OLPEDGE-1659

Signed-off-by: Liubov Didkivska <ext-liubov.didkivska@here.com>

### CODE REVIEW

File: olp-cpp-sdk-dataservice-read/src/ApiClientLookup.cpp [hunk 1]
```diff
@@ -57,7 +57,7 @@ ApiClientLookup::ApiClientResponse ApiClientLookup::LookupApi(
     client::OlpClientSettings settings) {
   repository::ApiCacheRepository repository(catalog, settings.cache);
 
-  if (options != OnlineOnly) {
+  if (options != OnlineOnly && options != CacheWithUpdate) {
```
Comment (andescu) [inline olp-cpp-sdk-dataservice-read/src/ApiClientLookup.cpp]: Why also CacheWithUpdate?

Comment (LiubovDidkivska) [reply olp-cpp-sdk-dataservice-read/src/ApiClientLookup.cpp]: Previously for CacheWithUpdate was splitted to CacheOnly and OnlineOnly. Now with OnlineOnly we are not updating cache. So for updating cache and getting data only online, even if we have valid stored cache we need additional fetch option. And to not brake api, I used CacheWithUpdate for this case.

File: olp-cpp-sdk-dataservice-read/tests/CatalogRepositoryTest.cpp [hunk 2]
```diff
@@ -183,9 +183,9 @@ TEST_F(CatalogRepositoryTest, GetLatestVersionOnlineOnlyFoundAndCacheWritten) {
         return boost::any{};
```
Comment (andescu) [inline olp-cpp-sdk-dataservice-read/tests/CatalogRepositoryTest.cpp]: Test name is then wrong, it says -> GetLatestVersionOnlineOnlyFoundAndCacheWritten
But you are not testing that the cache is written, you actually test that the cache is not written.

File: olp-cpp-sdk-dataservice-read/tests/CatalogRepositoryTest.cpp [hunk 3]
```diff
@@ -351,9 +351,9 @@ TEST_F(CatalogRepositoryTest, GetCatalogOnlineOnlyFoundAndCacheWritten) {
...
```
Comment (andescu) [inline olp-cpp-sdk-dataservice-read/tests/CatalogRepositoryTest.cpp]: Same here. Change test name.

**Pull request merged**